golem-worker-service-base/src/evaluator/evaluator_context.rs

```diff
@@ -7,23 +7,59 @@ use crate::worker_binding::{RequestDetails, WorkerDetail};
 use crate::worker_bridge_execution::RefinedWorkerResponse;
 use async_trait::async_trait;
 
-use golem_service_base::model::WorkerId;
-use golem_wasm_ast::analysis::AnalysedFunction;
+use golem_common::model::parse_function_name;
+use golem_service_base::model::{ComponentMetadata, FunctionParameter, FunctionResult, WorkerId};
 use golem_wasm_rpc::TypeAnnotatedValue;
-use std::fmt::Display;
+use std::fmt::{Display, Formatter};
 
 #[derive(Clone)]
 pub struct EvaluationContext {
     pub variables: Option<TypeAnnotatedValue>,
-    pub analysed_functions: Vec<AnalysedFunction>,
+    pub functions: Vec<Function>,
```

Comment:
@jdegoes  This is not fully what we discussed, but consider this as one step towards what we envisioned, while fixing a bug immediately

Reply:
Forward progress, I like it!